### ISSUE TITLE
CA1721 does not consider method/property accessibility

### ISSUE THREAD

**Issue by pakdev** [dotnet/roslyn-analyzers#1088]
#### Analyzer package

Microsoft.ApiDesignGuidelines.Analyzers

#### Analyzer

PropertyNamesShouldNotMatchGetMethods

#### Repro steps

```cs
public class Person
{
    // public property, private method
    public string Name
    {
        get { return "Roslyn"; }
    }

    private string GetName()
    {
        return Name;
    }

    // private property, public method (admittedly contrived)
    private int Age
    {
        get { return 30; }
    }

    public int GetAge()
    { 
        return Age;
    }
}
```

#### Expected behavior

Neither scenario above should cause CA1721 to fire because the rule states "The name of a **public** or **protected** member starts with 'Get' and otherwise matches the name of a **public** or **protected** property."

#### Actual behavior

Fires
